### PR TITLE
Include completions of the final word in fulltext search

### PR DESCRIPTION
**Pull request by crisptrutski**
### Description

This switches to a slightly lower level query type, so we can include completions of the final word.

~~We could do more to preserve support for operators like and, or, and not, but I think this is enough for a v1.~~ It turned out I needed to do some more clean up of the input stream, so just went ahead and got this working, as it was fun.

A big difference from the existing search is that we require all the terms, rather than some. Biasing towards search being fast, and typically this is what people will want.

### CODE REVIEW

**Comment by crisptrutski** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/metabase/metabase/47848?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#47848** <a href="https://app.graphite.dev/github/pr/metabase/metabase/47848?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#47846** <a href="https://app.graphite.dev/github/pr/metabase/metabase/47846?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @crisptrutski and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/metabase/search/postgres/index.clj [hunk 1]
```diff
@@ -100,15 +100,23 @@
...
       (t2/insert! pending-table entry))))
 
+(defn- to-tsquery-expr [search-term]
+  (let [in-word? (= search-term (str/trimr search-term))
+        trimmed (str/trim search-term)
+        words (str/split trimmed #"\s+")]
+    (str (str/join " & " words)
+         (when in-word?
+           ":*"))))
+
```
Comment (crisptrutski) [inline src/metabase/search/postgres/index.clj:110]: Here's the action.

**Review by sansolovyov — APPROVED** [review]
I think it would be nice to test output of `search-query`, so it's 1) easier to understand what are you going for 2) easier to improve

**Comment by crisptrutski** [comment]
> I think it would be nice to test output of `search-query`, so it's 1) easier to understand what are you going for 2) easier to improve

Yeah it's going to be crucial to have these tests. I haven't gone deep down this route yet as I need to figure out how to properly encapsulate the appdb state for these tests. Coming soon!

**Comment by crisptrutski** [comment]
@piranha how about these tests? https://github.com/metabase/metabase/blob/9bda9b513c5b7eed408d51b6beabcd901d5037d7/test/metabase/search/postgres/index_test.clj#L114-L140

**Pull request merged**